src/transform/EditTransform.js

```diff
@@ -0,0 +1,34 @@
+import './EditTransform.css'
+
+const CLASS = {
+  CONTAINER: 'pagelib_edit_section_link_container',
+  LINK: 'pagelib_edit_section_link',
+  PROTECTION: { UNPROTECTED: '', PROTECTED: 'page-protected', FORBIDDEN: 'no-editing' }
+}
+
+const DATA_ATTRIBUTE = { SECTION_INDEX: 'data-id', ACTION: 'data-action' }
+const ACTION_EDIT_SECTION = 'edit_section'
+
+/**
+ * @param {!Document} document
+ * @param {!number} index The zero-based index of the section.
+ * @return {!HTMLSpanElement}
+ */
+const newEditSectionLink = (document, index) => {
+  const container = document.createElement('span')
```

Comment:
I don't mind breaking the true link part out like you suggested but I think it's clearer to the client if we keep calling this a "link" or a "button." This makes a new component and that component is logically a link or a button. Thoughts?